design-documents/graph-ql/coverage/custom-attributes/custom-attributes-container.md

```diff
@@ -8,13 +8,42 @@ One workaround for "getting all fields" is based on schema introspection, it all
 
 # Proposed solution
 
-To account for dynamic nature of EAV attributes and the need of "getting all fields" in product search queries, `custom_attributes: [CustomAttribute]!` container will be introduced. 
+To account for dynamic nature of EAV attributes and the need of "getting all fields" in product search queries,we can introduce `custom_attributes: [CustomAttribute]!` container. 
 
 ```graphql
 type CustomAttribute {
```

Comment:
I thinks we have to add "type" opinion here to be able to render the appropriate template (datetime, text, multi-select, checkbox ...)

E.g.
```
"attributes":[
    {
      "attributeCode":"multiselect_attribute",
      "type":"multiselect",
      "value":[
        {
          "id":"21",
          "value":"Option 1"
        },
        {
          "id":"22",
          "value":"Option 2"
        }
      ]
    }
  ]
```

Reply:
Attributes are going to be displayed as information and not as fields you can control, so I don't think it's necessary.

Reply:
it will be useful to render field properly on U without complex logic. E.g. for "Date" field - "10-11-2020"

Reply:
yes, they are needed for the forms

Reply:
We have this data already available in the scope of `customAttributeMetadata` query. Since update frequency for attribute metadata and product attribute values is different, it makes sense to separate them to avoid excessive payload size.